docs/Migrating_LLB_to_a_PPL_based_Custom_Device.md

```diff
@@ -0,0 +1,74 @@
+
+* Certain VeriStand Custom Device APIs (like the NI VeriStand Custom Device Channel APIs) are using the Global Data References 
+
+* Mutation code might be required for individual custom devices if custom data types (for instance, clusters) are used internally and stored within the System Definition file, if we want to ensure backwards compatibility. 
+### Implementing a PPL based Custom Device
+
+For creating a new packed project library based Custom Device you will begin by building a template project. You can find the necessary template here: [VeriStand Custom Device Wizard](https://github.com/ni/niveristand-custom-device-wizard/releases). The new template will be available starting with VeriStand 2021 (Custom Device Wizard LabView 2021 support 21.0). 
+### Migrating an existing Custom Device
+
+First step for migrating an existing Custom Device is to make some changes regarding the global data references if you use certain VeriStand Custom Device APIs.
+For performance reasons, channel values are stored in VeriStand as a single block of data - i.e. as an array of double values. To be able to access a value element corresponding to a given channel, VeriStand is using Global Variables to pass pointer information, from the engine, to the calling APIs. While this works well for custom devices packed in LLBs, the same does not apply for packed project libraries, due to namespacing. Essentially, when compiling a PPL, a separate (namespaced) copy of the global variable is created and included with it. In turn, it cannot be used anymore for data transfer between the VeriStand engine and the running custom device and would result in a runtime error. Custom devices that do not use this kind of APIs are not affected by this issue. For instance, mainly Inline custom devices ( and derivatives of this type) are affected, since custom devices of Async type are using RT FIFOs to access channel values.
+To be able to mitigate this problem, we need to implement an  alternative way to access the values within these global variables. 
+This code needs to be incapsulated within a single VI ("Initialize Global Variables"). It has to be called only once from within the target custom device and, for an Inline type, it would have to be called specifically in the "Read Data from HW" state of the "RT Driver" due to the specifics of the initialization of the VeriStand engine.
```

Comment:
```suggestion
If you use certain VeriStand Custom Device APIs, which is the case for an inline custom device, the first step to migrate an existing Custom Device would include changes regarding how the global data references are initialized.

For performance reasons, channel values are stored in VeriStand as a single block of data - i.e. as an array of double values. To be able to access a value element corresponding to a given channel, VeriStand is using Global Variables to pass pointer information, from the engine, to the calling APIs. While this works well for custom devices packed in LLBs, the same does not apply for packed project libraries, due to namespacing. Essentially, when compiling a PPL, a separate (namespaced) copy of the global variable is created and included with it. In turn, it cannot be used anymore for data transfer between the VeriStand engine and the running custom device and would result in a runtime error. To be able to mitigate this problem, we need to implement an alternative way to access the values within these global variables.

The initialization code below needs to be incapsulated within a subVI ("Initialize Global Variables"). It has to be called only once from within the target custom device and, for an Inline type, it would have to be called specifically in the "Read Data from HW" case of the "RT Driver" due to how the VeriStand engine initializes.
```